extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
     width="480" height="270"></amp-youtube>
 <figcaption>Watch this video for an introduction to amp-bind.</figcaption></figure>
 
-### A simple example
+`amp-bind` has three main concepts:
 
-In the following example, tapping the button changes the `<p>` element's text from "Hello World" to "Hello amp-bind".
+1. [State](#state): A document-scope, mutable JSON state. State variables update in response to user actions. `amp-bind` does not evaluate expressions on page load. Visual elements should have their default "state" defined and not rely `amp-bind` for initial render.
+2. [Expressions](#expressions): JavaScript-like expressions that can reference the **state**.
+3. [Bindings](#bindings): Special attributes that link an element's property to a **state** via an **expression**. A property is bound by wrapping it inside brackets, in the form of `[property]`.
+
+### Basic example
+
+[example preview="inline" playground="true" imports="amp-bind"]
 
 ```html
 <p [text]="'Hello ' + foo">Hello World</p>
 
-<button on="tap:AMP.setState({foo: 'amp-bind'})">Say "Hello amp-bind"</button>
+<button on="tap:AMP.setState({foo: 'Interactivity'})">
+  Say "Hello Interactivity"
+</button>
 ```
 
-{% call callout('Note', type='note') %}
-For performance and to avoid the risk of unexpected content jumping, `amp-bind` does not evaluate expressions on page load. This means that the visual elements should be given a default state and not rely `amp-bind` for initial render.
-{% endcall %}
+[/example]
 
-### How does it work?
+In the example above:
 
-`amp-bind` has three main components:
+- The **state** begins as empty.
+- It has a single **binding** to `[text]`, the text content of a node, on the `<p>` element.
+- The `[text]` value contains the **expression**, `'Hello ' + foo`. This expression concatenates the string 'Hello ' and the value of the **state variable** foo.
```

Comment:
Like this change too, 'string literal' while accurate is overboard for the example.